### PR TITLE
Enable Managed Transport by default

### PR DESCRIPTION
**Pull request by pjbgf**
- Enable Managed Transport by default
- Update to latest source-controller to benefit from Managed Transport improvements:
  - Panic recovery for Git operations.
  - Improved SSH connection management without use of caching.
  - Enforce context timeout for managed SSH.
  - Remove dependency to callback functions.
  - Add support for hashed known_hosts.
 - Fixes regression introduced by #330 by which IAC will fail to update a push branch when the push branch already exists and the checkout branch is ahead.

Test image: `ghcr.io/fluxcd/image-automation-controller:rc-48bcca59`

### CODE REVIEW

**Comment by pjbgf** [comment]
Encountered a few issues whilst updating source-controller which resulted in fixing a regression and creating a new issue for the follow-up task of enabling the tests to run against the [Managed Transport](https://github.com/fluxcd/image-automation-controller/issues/370) version of the controller.

**Comment by pjbgf** [comment]
Manually tested on both GitHub and DevOps.

image
image

File: controllers/imageupdateautomation_controller.go [hunk 1]
```diff
@@ -732,7 +735,28 @@ var errRemoteBranchMissing = errors.New("remote branch missing")
...
+		// However, not setting them may lead to git2go panicing.
+		callbacks = managed.RemoteCallbacks()
+	}
+
 	branchRef := fmt.Sprintf("origin/%s", branch)
+	// Force the fetching of the remote branch.
+	err = origin.Fetch([]string{branch}, &libgit2.FetchOptions{
+		RemoteCallbacks: callbacks,
+	}, "")
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (hiddeco) [reply controllers/imageupdateautomation_controller.go:756]: I would ignore the error for now, as I am not confident on the full impact on the further code below by doing a quick inspection.

[the remark this answers is not in the document]
Comment (aryan9600) [reply controllers/imageupdateautomation_controller.go:756]: Okay, upon testing, it looks like if the branch doesn't exist, the error returned is nil, so no need to check for `libgit2.ErrorCodeNotFound`.

**Pull request merged**